dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/data/PutJobData.kt

```diff
@@ -112,20 +120,34 @@ data class PutJobData(
     override fun getObjectChannelBuilder(prefix: String): Ds3ClientHelpers.ObjectChannelBuilder =
         EmptyChannelBuilder(FileObjectPutter(prefixMap.get(prefix)!!.parent), prefixMap.get(prefix)!!)
 
-    private fun dataToDs3Objects(item: Pair<String, Path>): Sequence<Pair<Ds3Object, Path>> {
-        val localDelim = item.second.fileSystem.separator
-        val parent = item.second.parent
-        val paths = item.second.toFile().walk(FileWalkDirection.TOP_DOWN)
-                    .filter { !(it.isDirectory && Files.list(it.toPath()).use { f -> f.findAny().isPresent }) }
-                    .map {
-                        Ds3Object(if (it.isDirectory) {
-                            targetDir + parent.relativize(it.toPath()).toString().replace(localDelim, "/") + "/"
-                        } else {
-                            targetDir + parent.relativize(it.toPath()).toString().replace(localDelim, "/")
-                        }, if (Files.isDirectory(it.toPath())) 0L else it.length())
-                    }
-                .map { Pair(it, item.second) }
-        return paths
+    private fun pathToDs3Object(path: Path): Sequence<Pair<Ds3Object, Path>> {
+        return path.toFile().walk(FileWalkDirection.TOP_DOWN)
+            .filter(::rejectEmptyDirectory)
```

Comment:
This is way cleaner.  It really communicates the intention behind what each of the steps is doing.